app/controllers/view_components_controller.rb

```diff
@@ -32,7 +32,8 @@ def previews
       template = @render_args[:template]
       locals = @render_args[:locals]
       opts = {}
-      opts[:layout] = layout if layout.present? || layout == false
+      opts[:layout] = "previews"
+      @host_application_layout = layout if layout.present? || layout == false
```

Comment:
It's not clear to me why we need to mess around with layouts to achieve what we need here. Can you share the reasons for this approach?

Reply:
We want to append some HTML for the source code example below every preview. My thinking was that this would be easiest archived if all previews are being rendered in the same layout where `yield` renders the preview HTML and below we render the HTML for the source code example. 
I initially intended this to work without the user having to add anything in the layout the preview gets rendered in. This didn't work and users now have to add `yield(:view_component_preview_source_code_example)` in their layout which adds the source code example. 

Given that a user anyways has to explicitly render the preview source code example in a layout in the host application we can also solve this using a render call of a partial. That way we don't have to touch any logic regarding layouts in the view components controller. Does this sound like a better approach or do you have any suggestion of how to solve this without a user having to change anything in their layout?

Reply:
Can we just add to `app/views/view_components/preview.html.erb` to append the source code after the preview? Maybe I'm missing something.

Reply:
This does not work for [preview templates](https://viewcomponent.org/guide/previews.html#preview-templates) for which the view files are stored in the host application.